### PR TITLE
Add plot directive to LearningCurve documentation.

### PR DESCRIPTION
**Pull request by navarretedaniel**
# LearningCurve

## Overview
The following changes were made to address the plot directive project referenced in #687:

1. Deleted `learning_curve.py` script previously used to generate the images

2. Deleted the original image files (`learning_curve_classifier.png`, `learning_curve_regressor.png`, and `learning_curve_clusterer.png`).

3. Updated `learning_curve.rst` with the new plot directive, including adding the `:context: close-figs and :alt:` options

## Results
As a result of these changes, the following images rendered identically to those (currently) found in production:

- [x] `learning_curve_classifier.png`

- [x] `learning_curve_regressor.png`

- [ ] `learning_curve_clusterer.png`

## Clustering Example
I had issue replicating the plot depicted in `learning_curve_clusterer.png`.

I believe this issue results from the fact that a `random_state` parameter can be applied (but wasn't) in `make_blobs()`, `KMeans()`, and `LearningCurve()`.  As a result, it's difficult to pinpoint which combination of `random_state` inputs would result in the exact replication of the original image.

Additionally, failing to assign a `random_state` parameter in any of the three functions referenced above will result in replication issues as a new learning curve is generated with each plot.

To ensure future replication, `random_state=42` was applied across each of the three functions in the clustering example.  The resulting plot/image is as follows:

lrandom_state=42

### CODE REVIEW

**Comment by pdamodaran** [comment]
@navarretedaniel - Thank you so much for working on this!  Your updates look great and thanks for cleaning up some of the wording and making sure the visualizer naming in the code follows our convention.  I don't think the change in the learning_curve_clusterer.png is an issue. I think this is ready to be merged in -  @lwgray or @Kautumn06 - do you have any additional feedback?

**Comment by lwgray** [comment]
@navarretedaniel As we talked about offline - This is a great PR and we will get to it asap.  Great Job!

File: docs/api/model_selection/learning_curve.rst [hunk 1]
```diff
@@ -23,39 +23,36 @@ Classification
...
+    X, y = load_game()
 
     # Encode the categorical data with one-hot encoding
-    X = pd.get_dummies(data[features])
-    y = data[target]
+    X = OneHotEncoder().fit_transform(X)
 
     # Create the learning curve visualizer
-    cv = StratifiedKFold(12)
+    cv = StratifiedKFold(n_splits=12, random_state=42)
```
Comment (lwgray) [inline docs/api/model_selection/learning_curve.rst]: I am unsure if random_state is necessary here... @Kautumn06 your input is needed

File: docs/api/model_selection/learning_curve.rst [hunk 2]
```diff
@@ -93,23 +85,25 @@ Clustering
...
     from sklearn.datasets import make_blobs
 
-    # Create a dataset of blobs
-    X, y = make_blobs(n_samples=1000, centers=5)
-
-    viz = LearningCurve(
-        KMeans(), train_sizes=sizes, scoring="adjusted_rand_score"
-    )
+    # Generate synthetic dataset with 5 random clusters
+    X, y = make_blobs(n_samples=1000, centers=5, random_state=42)
```
Comment (lwgray) [inline docs/api/model_selection/learning_curve.rst]: Great use of random_state, I think it is good to have figures that users can replicate

File: docs/api/model_selection/learning_curve.rst [hunk 3]
```diff
@@ -3,7 +3,7 @@
 Learning Curve
 ==============
 
-A learning curve shows the relationship of the training score vs the cross validated test score for an estimator with a varying number of training samples. This visualization is typically used two show two things:
+A learning curve shows the relationship of the training score versus the cross validated test score for an estimator with a varying number of training samples. This visualization is typically used two show two things:
```
Comment (lwgray) [inline docs/api/model_selection/learning_curve.rst]: good catch... thanks for spelling this out

File: docs/api/model_selection/learning_curve.rst [hunk 4]
```diff
@@ -23,39 +23,36 @@ Classification
...
-    # Specify the features of interest and the target
-    target = "outcome"
-    features = [col for col in data.columns if col != target]
+    # Load a classification dataset
+    X, y = load_game()
 
     # Encode the categorical data with one-hot encoding
-    X = pd.get_dummies(data[features])
-    y = data[target]
+    X = OneHotEncoder().fit_transform(X)
```
[the remark this answers is not in the document]
Comment (navarretedaniel) [reply docs/api/model_selection/learning_curve.rst]: File has been updated to include label encoding for `y`.

File: docs/api/model_selection/learning_curve.rst (hunk 1 above)
Comment (navarretedaniel) [reply docs/api/model_selection/learning_curve.rst]: `random_state=42` has been removed. 

The plot renders identically without this `random_state`.

**Comment by rebeccabilbro** [comment]
Hey there @navarretedaniel @lwgray @Kautumn06 @bbengfort it looks like this PR is failing due to an unrelated update to `numba` (version 0.43), which is a dependency of the `umap-learn` library we use in the `yellowbrick.text.UMAPVisualizer` &mdash; [details here](https://github.com/lmcinnes/umap/issues/209). I've addressed the issue (temporarily) by pegging our test requirements to older versions of `umap-learn` and `numba`.
